hazelcast/src/main/java/com/hazelcast/jet/impl/client/protocol/task/JetMessageTaskFactoryProvider.java

```diff
@@ -53,39 +52,48 @@ public JetMessageTaskFactoryProvider(NodeEngine nodeEngine) {
+        factories.put(JetGetJobMetricsCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetGetJobMetricsMessageTask(cm, node, con));
+        factories.put(JetGetJobIdsCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetGetJobIdsMessageTask(cm, node, con));
+        factories.put(JetJoinSubmittedJobCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetJoinSubmittedJobMessageTask(cm, node, con));
+        factories.put(JetGetJobSubmissionTimeCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetGetJobSubmissionTimeMessageTask(cm, node, con));
+        factories.put(JetGetJobConfigCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetGetJobConfigMessageTask(cm, node, con));
+        factories.put(JetResumeJobCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetResumeJobMessageTask(cm, node, con));
+        factories.put(JetExportSnapshotCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetExportSnapshotMessageTask(cm, node, con));
+        factories.put(JetGetJobSummaryListCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetGetJobSummaryListMessageTask(cm, node, con));
         factories.put(JetGetJobAndSqlSummaryListCodec.REQUEST_MESSAGE_TYPE,
-                toFactory(JetGetJobAndSqlSummaryListMessageTask::new));
+                (cm, con) -> new JetGetJobAndSqlSummaryListMessageTask(cm, node, con));
         factories.put(JetExistsDistributedObjectCodec.REQUEST_MESSAGE_TYPE,
-                toFactory(JetExistsDistributedObjectMessageTask::new));
+                (cm, con) -> new JetExistsDistributedObjectMessageTask(cm, node, con));
         factories.put(JetIsJobUserCancelledCodec.REQUEST_MESSAGE_TYPE,
-                toFactory(JetIsJobUserCancelledMessageTask::new));
-        factories.put(JetUploadJobMetaDataCodec.REQUEST_MESSAGE_TYPE, toFactory(JetUploadJobMetaDataTask::new));
-        factories.put(JetUploadJobMultipartCodec.REQUEST_MESSAGE_TYPE, toFactory(JetUploadJobMultipartTask::new));
+                (cm, con) -> new JetIsJobUserCancelledMessageTask(cm, node, con));
+        factories.put(JetUploadJobMetaDataCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetUploadJobMetaDataTask(cm, node, con));
+        factories.put(JetUploadJobMultipartCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetUploadJobMultipartTask(cm, node, con));
+        factories.put(JetAddJobStatusListenerCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetAddJobStatusListenerMessageTask(cm, node, con));
+        factories.put(JetRemoveJobStatusListenerCodec.REQUEST_MESSAGE_TYPE,
+                (cm, con) -> new JetRemoveJobStatusListenerMessageTask(cm, node, con));
     }
 
     @Override
     public Int2ObjectHashMap<MessageTaskFactory> getFactories() {
         return factories;
     }
-
-    private MessageTaskFactory toFactory(MessageTaskConstructor constructor) {
```

Comment:
what was wrong with `toFactory`?

Reply:
It can affect the performance slightly and in `DefaultMessageTaskFactoryProvider`, there is no such wrapping. AFAIK, Java doesn't have an AST manipulation API similar to `ExpressionBuilder` in C# and helper methods may create long invocation chains. JVM can perform some optimizations to effectively inline them, but there is no guarantee.

Reply:
ok, thanks for explanation. However, Jet operations should not be that often, that this could make measurable difference.